src/bindings/python/flux/cli/base.py

```diff
@@ -757,7 +758,11 @@ def __init__(self, prog, usage=None, description=None, exclude_io=False):
         self.exitcode = 0
         self.progress = None
         self.watcher = None
+        self.plugins = CLIPluginRegistry().load_plugins(prog)
         self.parser = self.create_parser(prog, usage, description, exclude_io)
+        group = self.parser.add_argument_group("Options provided by plugins")
+        for option in self.plugins.get_options():
+            group.add_argument(option.__dict__.pop("name"), **option.__dict__)
```

Comment:
Another spot where having explicit attributes in the CLIPluginOption class would be good. Then you could split the option name and kwargs into `option.name` and `option.kwargs` and this code would be cleaner as:
```python
        for option in self.plugins.get_options():
            group.add_argument(option.name, **option.kwargs)